assets/admin/editor-wizard/wizard.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+
+/**
+ * Wizard component.
+ *
+ * @param {Object}   props
+ * @param {Array}    props.steps        Array with the steps that will be rendered.
+ * @param {Function} props.onCompletion Callback to call when wizard is completed.
+ */
+const Wizard = ( { steps, onCompletion } ) => {
+	const [ currentStepNumber, setCurrentStepNumber ] = useState( 0 );
+	const [ data, setData ] = useState( {} );
+
+	const CurrentStep = steps[ currentStepNumber ];
+
+	const goToNextStep = () => {
+		if ( currentStepNumber + 1 < steps.length ) {
+			setCurrentStepNumber( currentStepNumber + 1 );
+		} else {
+			onCompletion( data );
+		}
+	};
+
+	return (
+		<div className={ 'sensei-editor-wizard' }>
+			<div className={ 'sensei-editor-wizard__body' }>
+				<CurrentStep
+					data={ data }
+					setData={ setData }
+					onCompletion={ onCompletion }
```

Comment:
Well, that's a good point! Probably `goToNextStep` might be enough. I like being more "explicit" with the `onCompletion` but sure it's not really relevant and we could go with `goToNextStep`.

I will update the comment in the sample "pattern" steps so that whoever implements them takes into account both approaches.